src/account/operations/transfer/high_level/sweep_outputs.rs

```diff
@@ -0,0 +1,242 @@
+            Some(strategy) => match &strategy.remainder_value_strategy {
+                RemainderValueStrategy::ReuseAddress | RemainderValueStrategy::ChangeAddress => gen_addr.await?.address,
+                RemainderValueStrategy::CustomAddress(account_address) => account_address.address.clone(),
+            },
+        };
+
+        Ok(address)
+    }
+
+    pub(crate) async fn find_nft_output(&self, nft_id: NftId) -> crate::Result<(OutputId, NftOutput)> {
+        let account = self.read().await;
+
+        let (output_id, output_data) = account
+            .unspent_outputs()
+            .iter()
+            .find(|(&output_id, output_data)| match &output_data.output {
+                Output::Nft(nft_output) => nft_output.nft_id().or_from_output_id(output_id) == nft_id,
+                _ => false,
+            })
+            .ok_or(Error::NftNotFoundInUnspentOutputs)?;
+
+        let nft_output = match &output_data.output {
+            Output::Nft(nft_output) => nft_output.clone(),
+            _ => unreachable!("We already checked that it's an nft output"),
+        };
+
+        Ok((*output_id, nft_output))
+    }
+
+    pub(crate) async fn find_alias_output(&self, alias_id: AliasId) -> crate::Result<(OutputId, AliasOutput)> {
+        let account = self.read().await;
+
+        let (output_id, output_data) = account
+            .unspent_outputs()
+            .iter()
+            .find(|(&output_id, output_data)| match &output_data.output {
+                Output::Alias(alias_output) => alias_output.alias_id().or_from_output_id(output_id) == alias_id,
+                _ => false,
+            })
+            .ok_or_else(|| Error::BurningFailed("Alias output not found".to_string()))?;
+
+        let alias_output = match &output_data.output {
+            Output::Alias(alias_output) => alias_output.clone(),
+            _ => unreachable!("We already checked that it's an alias output"),
+        };
+
+        Ok((*output_id, alias_output))
+    }
+
+    pub(crate) async fn sweep_address_outputs(
+        &self,
+        address: Address,
+        remainder_address: AddressWrapper,
+    ) -> crate::Result<()> {
+        let query_parameters = vec![
+            QueryParameter::Address(address.to_bech32(remainder_address.bech32_hrp())),
+            QueryParameter::HasExpirationCondition(false),
+            QueryParameter::HasTimelockCondition(false),
+            QueryParameter::HasStorageDepositReturnCondition(false),
+        ];
+
+        let alias_output_ids = self.client.aliases_output_ids(query_parameters.clone()).await?;
+        let basic_output_ids = self.client.output_ids(query_parameters.clone()).await?;
+        let nft_output_ids = self.client.nfts_output_ids(query_parameters).await?;
+
+        let alias_outputs = self.client.get_outputs(alias_output_ids).await?;
+        let basic_outputs = self.client.get_outputs(basic_output_ids).await?;
+        let nft_outputs = self.client.get_outputs(nft_output_ids).await?;
```

Comment:
I think this needs to happen recursively with the alias/nft ids of the new outputs and foundry outputs also need to be requested and destroyed

Reply:
I thought so at first, but then I thought ownership should be transitive, so if A own B and B owns C, then A should own B right? Unless I'm missing something though

Reply:
At the end the same ed25519 address owns it, but if it's not in the unlock condition, then you wouldn't get it

Reply:
Hmmm... I see what you're saying.